test/functional/specs/desktop/C14394.js

```diff
@@ -0,0 +1,62 @@
+import { t, Selector, ClientFunction } from "testcafe";
+import createNetworkLogger from "../../helpers/networkLogger";
+import getResponseBody from "../../helpers/networkLogger/getResponseBody";
+import { responseStatus } from "../../helpers/assertions/index";
+import fixtureFactory from "../../helpers/fixtureFactory";
+import testServerUrl from "../../helpers/constants/testServerUrl";
+
+const urlCollector = `${testServerUrl}/test/functional/sandbox/html/alloySdk.html`;
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title:
+    "C14394: When ID migration is enabled and no identity cookie is found but legacy AMCV cookie is found, the ECID will be sent on the request",
+  url: urlCollector,
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14394",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const apiCalls = ClientFunction(() => {
+  document.cookie =
+    "AMCV_53A16ACB5CC1D3760A495C99%40AdobeOrg=77933605%7CMCIDTS%7C18290%7CMCMID%7C16908443662402872073525706953453086963%7CMCAAMLH-1580857889%7C9%7CMCAAMB-1580857889%7CRKhpRz8krg2tLO6pguXWp5olkAcUniQYPHaMWWgdJ3xzPWQmdj0y%7CMCOPTOUT-1580260289s%7CNONE%7CvVersion%7C4.5.1";
+
+  window.alloy("configure", {
+    configId: "9999999",
+    orgId: "53A16ACB5CC1D3760A495C99@AdobeOrg",
+    edgeBasePath: window.edgeBasePath,
+    debugEnabled: true,
+    idMigrationEnabled: true
+  });
+});
+
+test("Test C14394: When ID migration is enabled and no identity cookie is found but legacy AMCV cookie is found, the ECID will be sent on the request", async () => {
+  await apiCalls();
+
+  // NOTE: I assume a click is needed to get a response body. Putting the same event call in apiCalls() results in an undefined response body.
+  await t.click(Selector("#event-button"));
+
+  await responseStatus(networkLogger.edgeEndpointLogs.requests, 200);
+  await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
+
+  const request = JSON.parse(
+    networkLogger.edgeEndpointLogs.requests[0].request.body
+  );
+
+  await t
+    .expect(request.xdm.identityMap.ECID[0].id)
+    .eql("16908443662402872073525706953453086963");
+
+  const response = JSON.parse(
+    getResponseBody(networkLogger.edgeEndpointLogs.requests[0])
+  );
+
+  await t.expect(response.handle[0].type).eql("identity:result");
+  await t
+    .expect(response.handle[0].payload[0].id)
+    .eql("16908443662402872073525706953453086963");
```

Comment:
This is asserting that the first payload of the first handle will have the identity result. But if payloads or handles are in a different order, we really shouldn't fail the test, because it's still a proper Konductor response and everything will still behave properly. Instead, I would recommend we search through `response.handle` looking for any handles with the `identity:result` type. Then inside those handles we look for the first payload where `namespace.code` is `ECID`. Then check that the ID on that payload is correct. I suspect that searching the handles and their payloads will come up a lot, so it would be useful to create a method like `getPayloadsByType` just like we did in the Alloy source itself: https://github.com/adobe/alloy/blob/51d0f23286d11a6aff20597ad384a93a94dfeb6c/src/core/createResponse.js#L35-L39